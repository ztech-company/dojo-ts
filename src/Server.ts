import express, { Router } from 'express';

export class Server {
  readonly app = express();

  constructor(private readonly port: number, routes: Router[]) {
    routes.forEach((route) => this.app.use(route));
  }

  start() {
    this.app.listen(this.port, () => {
      console.log(`Example app listening at http://localhost:${this.port}`);
    });
  }
}
