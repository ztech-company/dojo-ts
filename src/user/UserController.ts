import { Request, Response, Router } from 'express';

export class UserController {
  readonly router = Router();

  constructor() {
    this.router.post('/users', this.addUser.bind(this));
    this.router.get('/users', this.getUsers.bind(this));
  }

  addUser(req: Request, res: Response): void {
    res.json({
      name: req.body.name,
      cpf: req.body.cpf,
    });
  }

  getUsers(req: Request, res: Response): void {
    res.json([
      {
        name: 'Fulano',
        cpf: '00000000000',
      },
      {
        name: 'Ciclano',
        cpf: '11111111111',
      },
    ]);
  }
}
