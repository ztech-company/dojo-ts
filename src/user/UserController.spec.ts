import request from 'supertest';
import { Server } from '../Server';
import { UserController } from './UserController';

describe('UserController', () => {
  const userController = new UserController();
  const routes = [userController.router];
  const server = new Server(0, routes);

  describe('GET /users', () => {
    it('should list user', () => {
      return request(server.app)
        .get('/users')
        .expect(200)
        .expect((res: Response) => {
          expect(res.body).toMatchSnapshot();
        });
    });
  });
});
