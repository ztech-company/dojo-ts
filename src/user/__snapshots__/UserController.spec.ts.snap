// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserController GET /users should list user 1`] = `
Array [
  Object {
    "cpf": "00000000000",
    "name": "Fulano",
  },
  Object {
    "cpf": "11111111111",
    "name": "Beltrano",
  },
]
`;
